ideascube/serveradmin/catalog.py

```diff
@@ -831,14 +832,25 @@ def _progress(*args):
                     self._progress(' {}'.format(remote.name), *args)
 
                 # TODO: Verify the download with sha256sum? Crypto signature?
-                try:
-                    urlretrieve(remote.url, tmppath, reporthook=_progress)
-                except ConnectionError:
-                    print("Warning: Impossible to connect to the remote"
-                          " {remote.name}({remote.url}).\n"
-                          "Continue anyway without this remote."
-                          .format(remote=remote))
-                    continue
+                urlparsed = urlparse(remote.url)
+                if urlparsed.scheme in ['file', '']:
+                    try:
+                        shutil.copyfile(urlparsed.path, tmppath)
+                    except Exception as error:
+                        print("Warning: Impossible to copy the catalog file"
+                              " {remote.name}({remote.url}).\n{error}\n"
+                              "Continue anyway without this remote."
+                              .format(remote=remote, error=error))
+                        continue
+                else:
+                    try:
+                        urlretrieve(remote.url, tmppath, reporthook=_progress)
+                    except ConnectionError:
+                        print("Warning: Impossible to connect to the remote"
+                            " {remote.name}({remote.url}).\n"
+                            "Continue anyway without this remote."
+                            .format(remote=remote))
```

Comment:
You broke the proper indentation of the string here.

This is such an offense to good taste that I must immediately refuse this pull request in the most obnoxious way possible.

Just kidding, but try to be mindful about that in the future. :stuck_out_tongue_winking_eye: